lib/grammar.pegjs

```diff
@@ -228,10 +249,16 @@ import_mixin_stmt = 'import' __ first:ident _ rest:(',' _ ident _)* 'from' __ na
   return new Ast.ImportStmt.Mixin([first].concat(take(rest, 2)), name, in_params);
 }
 
-function_def = modifiers:query_modifiers? _ type:('query' / 'action') __ name:ident _ args:function_param_decl_list _ annotations: ( _ annotation _)* ';' _ {
+function_name_list = first:ident _ rest:(',' _ ident _)* {
+    return [first].concat(take(rest, 2))
+}
+function_def = modifiers:query_modifiers? _ type:('query' / 'action') __ name:ident _ extends_:('extends' _ function_name_list _ )? args:function_param_decl_list _ annotations: ( _ annotation _)* ';' _ {
```

Comment:
You must use `__` after `'extends'` otherwise `extendsfoo` would parse as `extends foo`